src/gateway/routers/Aggregator.sol

```diff
@@ -218,4 +275,14 @@ contract Aggregator is Auth, IAggregator {
     function votes(bytes32 messageHash) external view returns (uint16[8] memory) {
         return messages[messageHash].votes;
     }
+
+    /// @inheritdoc IAggregator
+    function estimate(bytes calldata payload) external view returns (uint256 estimated) {
+        bytes memory proof = abi.encodePacked(uint8(MessagesLib.Call.MessageProof), keccak256(payload));
+        uint256 destChainCost = gasService.estimate(payload);
+
+        for (uint256 i; i < routers.length; i++) {
+            estimated += RouterLike(routers[i]).estimate(i == PRIMARY_ROUTER_ID - 1 ? payload : proof, destChainCost);
```

Comment:
would not intro `destChainCost` as a var since `gasService.estimate(payload)` is only called once within the function

Reply:
This is aggregator :D This will be deleted